File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.py
class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        minimum = set()
        result = list()

        for row in matrix:
            minimum.add(min(row))

        for col in zip(*matrix):
            target = max(col)
            if target in minimum:
                result.append(target)

        return result

File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.java
class Solution {
    public List<Integer> luckyNumbers (int[][] matrix) {
        Set<Integer> min = new HashSet<>();
        List<Integer> result = new ArrayList<>();

        for (int i = 0; i < matrix.length; i++) {
            int target = Integer.MAX_VALUE;
            for (int j = 0; j < matrix[0].length; j++) {
                target = Math.min(target, matrix[i][j]);
            }
            min.add(target);
        }

        for (int i = 0; i < matrix[0].length; i++) {
            int target = 0;
            for (int j = 0; j < matrix.length; j++) {
                target = Math.max(target, matrix[j][i]);
            }
            if (min.contains(target)) result.add(target);
        }

        return result;
    }
}